File: 16_6_22/a01.m
% Pies sin dice
clear; clc;

pie = @(a, b, a_b) a + b - a_b;

a = .1;
b = .2;
a_b = .05;

disp(pie(a, b, a_b))

% white and black
with_rem = @(b, w) factorial(w) / (factorial(1)*factorial(w-1)) * ...
    factorial(b) / (factorial(1)*factorial(b-1)) / ...
    (factorial(w+b) / (factorial(2)*factorial(w+b-2)));
sin_rem = @(b, w) factorial(2) / (factorial(1)*factorial(1)) * ...
    w*b / (w+b)^2;


b = 4;
w = 3;

disp("with " + with_rem(b, w))
disp("sin " + sin_rem(b, w))

File: 16_6_22/a01.py
# Pies and dice
def pie(a, b, a_b):   # Priciple of Inclusion and Exclusion, sållprincipen
    return a + b - a_b


a = .1
b = .2
a_b = .05

print(pie(a, b, a_b))


# Draws and yaws
import math


def with_removal(b, w):
    combinations = math.factorial(2) / (math.factorial(1)*math.factorial(1))
    # chance of one of each
    return combinations*b*w / (b+w)**2


def sin_removal(b, w):
    white_combinations = math.factorial(w) / (math.factorial(1) * math.factorial(w-1))
    black_combinations = math.factorial(b) / (math.factorial(1) * math.factorial(b-1))
    all_combinations = math.factorial(w+b) / (math.factorial(2) * math.factorial(b+w-2))
    return white_combinations*black_combinations/all_combinations


b = 4   # black balls
w = 3   # white balls

print("with", with_removal(b, w))
print("sin", sin_removal(b, w))
